### PR TITLE
xkbcommon: fix single-profile build

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **xkbcommon/***

fixes conan-io/conan-center-index#20542

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6533284561)

**Comment by jwillikers** [comment]
Hmmm, I can't seem to cross-compile the recipe with this change.

```
Build-time dependency wayland-protocols_build found: NO (tried pkgconfig and cmake)
Build-time dependency wayland-scanner_build found: NO (tried pkgconfig and cmake)

../src/meson.build:534:12: ERROR: Problem encountered: The Wayland xkbcli programs require wayland-client >= 1.2.0, wayland-protocols >= 1.7 which were not found.
You can disable the Wayland xkbcli programs with -Denable-wayland=false.
```

**Comment by ericLemanissier** [comment]
what does `/home/spaceim/.conan2/p/b/xkbco......../build-release/meson-logs/meson-log.txt` say ?

**Comment by jwillikers** [comment]
```
 799   │ Run-time dependency wayland-client found: YES 1.22.0
 800   │ Pkg-config binary for 0 is not cached.
 801   │ Pkg-config binary missing from cross or native file, or env var undefined.
 802   │ Trying a default Pkg-config fallback at pkg-config
 803   │ Found pkg-config: /usr/bin/pkg-config (2.0.3)
 804   │ Determining dependency 'wayland-protocols_BUILD' with pkg-config executable '/usr/bin/pkg-config'
 805   │ env[PKG_CONFIG_PATH]: 
 806   │ env[PKG_CONFIG_SYSROOT_MAP]: /usr/local/company-yocto-sdk/sysroots/cortexa53-crypto-company-linux
 807   │ env[PKG_CONFIG]: /home/jordan/.conan/data/pkgconf/2.0.3/_/_/package/77f233048f15709a5b7952a12eead7dd5a94d613/bin/pkgconf
 808   │ -----------
 809   │ Called: `/usr/bin/pkg-config --modversion wayland-protocols_BUILD` -> 1
 810   │ stderr:
 811   │ Package wayland-protocols_BUILD was not found in the pkg-config search path.
 812   │ Perhaps you should add the directory containing `wayland-protocols_BUILD.pc'
 813   │ to the PKG_CONFIG_PATH environment variable
 814   │ Package 'wayland-protocols_BUILD', required by 'virtual:world', not found
 815   │ -----------
...
       │ Build-time dependency wayland-protocols_build found: NO (tried pkgconfig and cmake)
 849   │ Pkg-config binary for 0 is cached.
 850   │ Determining dependency 'wayland-scanner_BUILD' with pkg-config executable '/usr/bin/pkg-config'
 851   │ env[PKG_CONFIG_PATH]: 
 852   │ env[PKG_CONFIG_SYSROOT_MAP]: /usr/local/company-yocto-sdk/sysroots/cortexa53-crypto-company-linux
 853   │ env[PKG_CONFIG]: /home/jordan/.conan/data/pkgconf/2.0.3/_/_/package/77f233048f15709a5b7952a12eead7dd5a94d613/bin/pkgconf
 854   │ -----------
 855   │ Called: `/usr/bin/pkg-config --modversion wayland-scanner_BUILD` -> 1
 856   │ stderr:
 857   │ Package wayland-scanner_BUILD was not found in the pkg-config search path.
 858   │ Perhaps you should add the directory containing `wayland-scanner_BUILD.pc'
 859   │ to the PKG_CONFIG_PATH environment variable
 860   │ Package 'wayland-scanner_BUILD', required by 'virtual:world', not found
 861   │ -----------
```

You still need to set `build.pkg_config_path` for Meson to find the pkg-config files in the build context. When cross-compiling, Meson will only look in `build.pkg_config_path` for pkg-config dependencies where `native=true`.

**Comment by jwillikers** [comment]
That works! 👍

File: recipes/xkbcommon/all/conanfile.py [hunk 1]
```diff
@@ -152,8 +144,7 @@ def package_info(self):
             if not self._has_build_profile:
                 self.cpp_info.components["xkbcli-interactive-wayland"].requires.append("wayland-protocols::wayland-protocols")
 
-        if Version(self.version) >= "1.0.0":
-            self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
+        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
```
Comment (jwillikers) [inline recipes/xkbcommon/all/conanfile.py:147]: I don't think this is even necessary anymore. The `bin` directory should get added automatically to `PATH`.

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@SpaceIm could you please try this recipe to test if it fixes the problem you face when building xkbcommon in a single profile context?

**Comment by SpaceIm** [comment]
> @SpaceIm could you please try this recipe to test if it fixes the problem you face when building xkbcommon in a single profile context?

I'll do when I have time. But I don't understand what you mean by single profile context. I don't use single profile. My issue comes from a build triggered from build context (with 2 profiles).

File: recipes/xkbcommon/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/xkbcommon/all/conanfile.py:147]: It's necessary for legacy generators.

**Comment by ericLemanissier** [comment]
I mean "in the build context"

**Comment by SpaceIm** [comment]
I'll try, but I don't understand how it could fix anything. `_has_build_profile()` is always True (or should, otherwise it's a bug) in conan v2, irrespective of context.

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6682411361)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6682411361)

**Comment by ericLemanissier** [comment]
I could reproduce the error with the following conanfile.txt:
```
[tool_requires]
xkbcommon/1.6.0
```
and running command `conan install . -b xkbcommon/*`

meson-log.txt shows the failure to detect wayland-client:

```
Determining dependency 'wayland-client' with pkg-config executable '/home/eric/.conan2/p/pkgco96229fb471f65/p/bin/pkgconf'
env[PKG_CONFIG]: /home/eric/.conan2/p/pkgco96229fb471f65/p/bin/pkgconf
env[PKG_CONFIG_PATH]: /home/eric/.conan2/p/b/xkbco79b95ed7858f7/b/build-release/conan
-----------
Called: `/home/eric/.conan2/p/pkgco96229fb471f65/p/bin/pkgconf --modversion wayland-client` -> 1
stderr:
Package wayland-client was not found in the pkg-config search path.
Perhaps you should add the directory containing `wayland-client.pc'
to the PKG_CONFIG_PATH environment variable
Package 'wayland-client', required by 'virtual:world', not found
-----------
CMake binary for 1 is cached.
Preliminary CMake check failed. Aborting.
Run-time dependency wayland-client found: NO (tried pkgconfig and cmake)
```
I could fix the detection issue by patching the wayland-client.pc file name in meson.build, but then link fails with:
```
FAILED: xkbcli-interactive-wayland
gcc  -o xkbcli-interactive-wayland xkbcli-interactive-wayland.p/meson-generated_xdg-shell-protocol.c.o xkbcli-interactive-wayland.p/tools_interactive-wayland.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -Wl,--start-group libtools-internal.a libxkbcommon.a /home/eric/.conan2/p/waylaf46d2984bd51a/p/lib/libwayland-client.a -lpthread -lm -lrt -Wl,--end-group
/usr/bin/ld: /home/eric/.conan2/p/waylaf46d2984bd51a/p/lib/libwayland-client.a(connection.c.o): in function `wl_closure_invoke':
connection.c:(.text+0x249a): undefined reference to `ffi_type_pointer'
/usr/bin/ld: connection.c:(.text+0x25b2): undefined reference to `ffi_type_pointer'
/usr/bin/ld: connection.c:(.text+0x25e8): undefined reference to `ffi_type_void'
/usr/bin/ld: connection.c:(.text+0x2602): undefined reference to `ffi_prep_cif'
/usr/bin/ld: connection.c:(.text+0x2630): undefined reference to `ffi_call'
/usr/bin/ld: connection.c:(.text+0x26c3): undefined reference to `ffi_type_sint32'
/usr/bin/ld: connection.c:(.text+0x26db): undefined reference to `ffi_type_uint32'
collect2: error: ld returned 1 exit status
```
cf https://github.com/ericLemanissier/proof-of-conan/actions/runs/6682411361/job/18157750381#step:7:2284
this error happens both in build context and in host context.
I don't have more time available for this issue, so I'm closing this.

**Pull request closed without merging**